app/scripts/directives/processTemplateDialog.js

```diff
@@ -124,5 +234,82 @@
     function instantiateTemplate() {
       $scope.$broadcast('instantiateTemplate');
     }
+
+    function filterForKeywords (searchText, items) {
+      return KeywordService.filterForKeywords(items, ['name', 'tags'], KeywordService.generateKeywords(searchText));
+    }
+
+    function filterChange (filters) {
+      // only use filters which have a filter criteria 'value'
+      // prevents applying an empty keyword filter
+      // TODO: can remove the following line of code after angular-patternfly issue is fixed:
+      //   https://github.com/patternfly/angular-patternfly/issues/509
+      filters = _.filter(filters, 'value');
+
+      ctrl.filterConfig.appliedFilters = filters;
+      filterItems();
+    }
+
+    function filterItems() {
+      ctrl.filteredItems = ctrl.catalogItems;
+      if (ctrl.filterConfig.appliedFilters && ctrl.filterConfig.appliedFilters.length > 0) {
+        _.each(ctrl.filterConfig.appliedFilters, function(filter) {
+          ctrl.filteredItems = filterForKeywords(filter.value, ctrl.filteredItems);
+        });
+      }
+
+      updateFilterControls();
+
+      // Deselect the currently selected template if it was filtered out
+      if (!_.includes(ctrl.filteredItems, ctrl.selectedTemplate)) {
+        ctrl.templateSelected();
+      }
+    }
+
+    function updateFilterControls() {
+      $timeout(function() {
```

Comment:
Always have to ask if `$` is needed/should be used or if an "angular native" way can be achieved via template bindings.

Reply:
I've removed the $timeout as it doesn't appear to be necessary.